servant-server/src/Servant/Server/Internal.hs

```diff
@@ -232,6 +259,75 @@ instance
 
   route Proxy = methodRouterHeaders methodDelete (Proxy :: Proxy ctypes) ok200
 
+-- | Authentication in Strict mode.
+instance
+#if MIN_VERSION_base(4,8,0)
+         {-# OVERLAPPABLE #-}
+#endif
+         (AuthData authdata , HasServer sublayout) => HasServer (AuthProtect authdata (usr :: *) 'Strict :> sublayout) where
+
+    type ServerT (AuthProtect authdata usr 'Strict :> sublayout) m = AuthProtected authdata usr (usr -> ServerT sublayout m) 'Strict
+
+    route _ subserver = WithRequest $ \req ->
+        route (Proxy :: Proxy sublayout) $ do
+            -- Note: this may perform IO for each attempt at matching.
+            rr <- routeResult <$> subserver
+
+            case rr of
+                -- Successful route match, so we extract the author-provided
+                -- auth data.
+                Right authProtectionStrict ->
```

Comment:
Would there be less drift to the right margin if we used `MaybeT IO` as the monad here, and then just checked for `Nothing` once at the end?